pinot-core/src/test/java/org/apache/pinot/queries/DistinctQueriesTest.java

```diff
@@ -170,163 +199,360 @@ private ImmutableSegment createSegment(int index, List<GenericRow> records)
     return ImmutableSegmentLoader.load(new File(INDEX_DIR, segmentName), ReadMode.mmap);
   }
 
+  @Test
+  public void testSingleColumnDistinctInnerSegment() {
```

Comment:
We should separate this into multiple individual unit tests.

Reply:
Split into `testSingleColumnDistinctOnlyInnerSegment` and `testSingleColumnDistinctOrderByInnerSegment`. I don't think we need to split further because that is not as easy to manage